src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
I still fear that in async_wait() errors may pile up because with bad timing an slow io error might only return an error when the whole operation is already in the network / ssh pipe.

But maybe that is not really going to happen. Also the pileup is saved in a compact form not as exceptions as i first thought. So maybe just a theoretical problem.